sympy/utilities/tests/test_lambdify.py

```diff
@@ -1406,3 +1406,93 @@ def test_cupy_dotproduct():
         f3(1, 2, 3) == \
         f4(1, 2, 3) == \
         cupy.array([14])
+
+
+def test_lambdify_cse():
+    def dummy_cse(exprs):
+        return (), exprs
+
+    def minmem(exprs):
+        from sympy.simplify.cse_main import cse_minimize_memory
+        return cse(exprs, postprocess=cse_minimize_memory)
+
+    class Case:
```

Comment:
I would rather write this test using simple data structures and a loop in the test itself, rather than a class. Like

```py
for args, exprs, num_args in [...]:
    assert ...
```

Reply:
I view this as a tidy pure-python way of running the test. Do you feel strongly about this @asmeurer? What problem do you see with organizing it this way.

Reply:
I suggest we defer style related refactorings in our non-public test suite to a later PR for now (in the interest of actually getting the functionality into SymPy), does that sound OK to you @asmeurer?